doc/src/programmers-guide.md

```diff
@@ -1568,3 +1568,202 @@ Once a connection is established, you can use a combination of
+|  `port` | `port_num()` |  | The port to which to bind the socket.  If no port is specified, the operating system will choose an ephermeral port for the user. |
+
+For example:
+
+    %% erlang
+    PortNumber = 8080,
+    ok = socket:bind(Socket, #{family => inet, addr => any, port => PortNumber}),
+
+To listen for connections, use the `socket:listen/1` function:
+
+    %% erlang
+    ok = socket:listen(Socket),
+
+Once your socket is listening on an interface and port, you can wait to accept a connection from an incoming client using the `socket:accept/1` function.
+
+This function will block the current execution context (i.e., Erlang process) until a client establishes a TCP connection with the server:
+
+    %% erlang
+    {ok, ConnectedSocket} = socket:accept(Socket),
+
+> Note.  Many applications will spawn processes to listen for socket connections, so that the main execution context of your application is not blocked.
+
+#### Sending and Receiving Data
+
+Once you have a connected socket, you can send and receive data on that socket using the `socket:send/2` and `socket:recv/1` functions.  Like the  `socket:accept/1` function, these functions will block until data is sent to a connected peer (or until the data is written to operating system buffers) or received from a connected peer.
+
+For example,
+
+    %% erlang
+    case socket:recv(ConnectedSocket) of
+        {ok, Data} ->
+            {ok, _Rest} = socket:send(ConnectedSocket, Data);
+        {error, closed} ->
+            io:format("Client closed connection.~n");
+        Error ->
+            io:format("An error occurred waiting on a connected socket: ~p~n", [Error])
+    end.
+
+> Note.  The `socket:recv/1` function will block the current process until a packet has arrived or until the local or remote socket has been closed, or some other error occurs.
+
+### Client-side TCP Socket Programming
+
+To program using sockets on the client side, you can connect an opened socket to an address and port number using the `socket:connect/2` function, supplying a map that specifies the address and port number.
+
+This map may contain the following entries:
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| `family` | `inet` | | The address family.  (Currently, only `inet` is supported) |
+|  `addr` | `addr()` | | The address to which to bind. The `loopback` value will bind the socket to the loopback interface on the device. |
+|  `port` | `port_num()` |  | The port to which to bind the socket. |
+
+    %% erlang
+    ok = socket:connect(Socket, #{family => inet, addr => loopback, port => 44404})
+
+#### Sending and Receiving Data
+
+Once you have a connected socket, you can send and receive data on that socket using the `socket:send/2` and `socket:recv/1` functions.
+
+For example:
```

Comment:
I rearranged some stuff, but I think it may not satisfy this change request.  But I think the way it is, while slightly repetitive, should still work for users.